torch/fx/graph_module.py

```diff
@@ -82,8 +87,7 @@ def forward(self, {', '.join(free_variables)}):
         }
         exec_with_source(self.code, gbls)
         cls = type(self)
-        for k, v in gbls.items():
-            setattr(cls, k, v)
+        setattr(cls, 'forward', gbls['forward'])
```

Comment:
exec with source and globals extraction now occurs in two places, should be a function like `generate_forward` that just returns the function implementation given the source and handles the globals creation.